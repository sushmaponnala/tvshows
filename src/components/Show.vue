<template>
  <div class="card mb-3">
    <div class="row no-gutters">
      <div class="col-md-4">
        <img :src="image" class="card-img" :alt="name" />
      </div>
      <div class="col-md-8 align-middle">
        <div class="card-body">
          <div class="text-truncate">
            <b-link class="text-primary" :to="'/showdetail/'+id">{{name}}</b-link>
          </div>
          <small class="card-text">
            Rating:
            <span class="text-muted">{{rating}}</span>
            <br />Genre(s):
            <span class="text-muted">{{genres.join(", ")}}</span>
          </small>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "Show",
  props: {
    id: Number,
    name: String,
    image: String,
    rating: Number,
    genres: Array
  }
};
</script>

<style scoped>
.card {
  max-width: 294px;
}
.card-body {
  padding: 0.75rem;
}
</style>
