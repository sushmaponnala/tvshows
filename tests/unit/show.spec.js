import { shallowMount, createLocalVue } from '@vue/test-utils'
import Show from '@/components/Show.vue'
import { BootstrapVue } from 'bootstrap-vue'

const localVue = createLocalVue()

localVue.use(BootstrapVue)

describe('Show.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(Show, {
      propsData: {
        id: 169,
        name: "Breaking Bad",
        image: "http://static.tvmaze.com/uploads/images/medium_portrait/0/2400.jpg",
        rating: 9.3,
        genres: [
          "Drama",
          "Crime",
          "Thriller"
        ]
      },
      localVue
    })
    expect(wrapper.element).toMatchSnapshot();
  })
})
