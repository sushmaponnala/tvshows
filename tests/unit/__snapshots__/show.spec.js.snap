// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show.vue renders correctly 1`] = `
<div
  class="card mb-3"
>
  <div
    class="row no-gutters"
  >
    <div
      class="col-md-4"
    >
      <img
        alt="Breaking Bad"
        class="card-img"
        src="http://static.tvmaze.com/uploads/images/medium_portrait/0/2400.jpg"
      />
    </div>
     
    <div
      class="col-md-8 align-middle"
    >
      <div
        class="card-body"
      >
        <div
          class="text-truncate"
        >
          <b-link-stub
            class="text-primary"
            event="click"
            routertag="a"
            target="_self"
            to="/showdetail/169"
          >
            Breaking Bad
          </b-link-stub>
        </div>
         
        <small
          class="card-text"
        >
          
          Rating:
          
          <span
            class="text-muted"
          >
            9.3
          </span>
           
          <br />
          Genre(s):
          
          <span
            class="text-muted"
          >
            Drama, Crime, Thriller
          </span>
        </small>
      </div>
    </div>
  </div>
</div>
`;
